src/gl.rs

```diff
@@ -87,17 +87,30 @@ pub fn read_pixels(x: GLint, y: GLint, width: GLsizei, height: GLsizei, format:
         _ => panic!("unsupported pixel_type for read_pixels"),
     };
 
-    let len = width * height * colors * depth;
-    let mut pixels: Vec<u8> = Vec::new();
-    pixels.reserve(len as usize);
+    return (width * height * colors * depth) as usize;
+}
+
+pub fn read_pixels_into_buffer(x: GLint, y: GLint, width: GLsizei, height: GLsizei,
+                               format: GLenum, pixel_type: GLenum,
+                               dst_buffer: &mut [u8], dst_buffer_length: usize) {
```

Comment:
You don't need to pass dst_buffer_length into the fn here - you can call dst_buffer.len()